### PR TITLE
BUG: Don't multiply sets during construction

### LINKED ISSUE #32582: Creating a column with a set replicates the set n times
#### Code Sample

If we try to define a dataframe using a dictionary containing a set, we get:

```python
pd.DataFrame({'a':{1,2,3}})

       a
0  {1, 2, 3}
1  {1, 2, 3}
2  {1, 2, 3}
```
#### Problem description

The set is being replicated `n` times, `n` being the length of the actual set. 
While defining a column with a set directly might not make a lot of sense given that they are by definition unordered collections, the behaviour  in any case seems clearly unexpected. 

#### Expected Output

In the case of a list, in order to obtain a single row containing a list, we would have to define a nested list, such as `pd.DataFrame({'a':[[1,2,3]]})`.
So similarly, with sets I would expect the same behaviour by defining the row with `pd.DataFrame({'a':[{1,2,3}]})`. 

In the case of a single set, even if the order is not guaranteed to be preserved, I'd see more reasonable the same output that we would obtain with:

```python

pd.DataFrame({'a':[1,2,3]})

   a
0  1
1  2
2  3
```
So:

```python
pd.DataFrame({'a':{1,2,3}})

   a
0  1
1  2
2  3
```

Where:
```python
pd.__version__
# '1.0.0'
```

### PR DESCRIPTION
**Pull request by dsaxton**
- [x] closes #32582
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: pandas/core/construction.py [hunk 1]
```diff
@@ -434,7 +434,9 @@ def sanitize_array(
             subarr = subarr.copy()
         return subarr
 
-    elif isinstance(data, (list, tuple)) and len(data) > 0:
+    elif isinstance(data, (list, tuple, set)) and len(data) > 0:
```
Comment (mroeschke) [inline pandas/core/construction.py]: Would it be appropriate to use `is_list_like(..., allow_sets=True)` here?

Comment (dsaxton) [reply pandas/core/construction.py]: Good question. It looks like that ends up interfering a bit with some of the other cases (`range` inputs that are handled differently below, as well `array.array` which causes a test to fail when I try to use that here).

Comment (mroeschke) [reply pandas/core/construction.py]: Alright sounds like it's not a good fit here (maybe make a comment here why we can't use `is_list_like`)

**Comment by WillAyd** [comment]
Hmm how is this type of construction actually useful? If anything this should probably just raise

**Comment by dsaxton** [comment]
> Hmm how is this type of construction actually useful? If anything this should probably just raise

Yeah, I could see that too. Looks like it does raise in the `Series` case so makes sense it would here too (or could treat sets as singletons, although that seems even less useful?)

**Comment by WillAyd** [comment]
If Series raises then I think that is even more motivation to raise here; this seems more likely to be a programmer error than intent

File: pandas/core/construction.py [hunk 2]
```diff
@@ -446,6 +446,8 @@ def sanitize_array(
         # GH#16804
         arr = np.arange(data.start, data.stop, data.step, dtype="int64")
         subarr = _try_cast(arr, dtype, copy, raise_cast_failure)
+    elif isinstance(data, set):
```
Comment (jreback) [inline pandas/core/construction.py]: can you use abc.Set

**Review by jreback — CHANGES_REQUESTED** [review]

**Review by WillAyd — APPROVED** [review]
Lgtm

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @dsaxton

**Pull request merged**